packages/hardhat-core/src/internal/artifacts.ts

```diff
@@ -96,21 +90,32 @@ export class Artifacts implements IArtifacts {
   }
 
   public async getArtifactPaths(): Promise<string[]> {
-    const paths = await glob(path.join(this._artifactsPath, "**/*.json"), {
-      ignore: [this._buildInfosGlob, this._dbgsGlob],
-    });
+    const buildInfosDir = path.join(this._artifactsPath, BUILD_INFO_DIR_NAME);
+
+    const paths = await getAllFilesMatching(
+      this._artifactsPath,
+      (f) =>
+        f.endsWith(".json") &&
+        !f.startsWith(buildInfosDir) &&
+        !f.endsWith(".dbg.json")
+    );
 
     return paths.sort();
   }
 
   public async getBuildInfoPaths(): Promise<string[]> {
-    const paths = await glob(this._buildInfosGlob);
+    const paths = await getAllFilesMatching(
+      path.join(this._artifactsPath, BUILD_INFO_DIR_NAME)
```

Comment:
To exactly match the previous implementation, this should filter by `f.endsWith(".json")`

Reply:
done